Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -265,6 +288,114 @@ - (void)urlSessionTask:(NSURLSessionTask *)sessionTask setState:(NSURLSessionTas
     SENTRY_LOG_DEBUG(@"SentryNetworkTracker finished HTTP span for sessionTask");
 }
 
+- (void)captureEvent:(NSURLSessionTask *)sessionTask
+{
+    NSHTTPURLResponse *myResponse = (NSHTTPURLResponse *)sessionTask.response;
+    NSNumber *responseStatusCode = [NSNumber numberWithLongLong:myResponse.statusCode];
+
+    if (!self.isCaptureFailedRequests || ![self containsStatusCode:myResponse.statusCode]) {
+        return;
+    }
+
+    if (![self isTargetMatch:sessionTask.currentRequest.URL
+                 withTargets:SentrySDK.options.failedRequestTargets]) {
+        return;
+    }
+
+    NSString *message = [NSString
+        stringWithFormat:@"HTTP Client Error with status code: %ld", (long)myResponse.statusCode];
+
+    SentryEvent *event = [[SentryEvent alloc] initWithLevel:kSentryLevelError];
+
+    SentryThreadInspector *threadInspector = SentrySDK.currentHub.getClient.threadInspector;
+    NSArray<SentryThread *> *threads = [threadInspector getCurrentThreads];
+
+    SentryException *sentryException = [[SentryException alloc] initWithValue:message
+                                                                         type:@"HTTP-ClientError"];
+    sentryException.mechanism =
+        [[SentryMechanism alloc] initWithType:@"SentryNetworkTrackingIntegration"];
+
+    if (threads.count > 0) {
+        for (SentryThread *thread in threads) {
+            if (nil != thread.current && [thread.current boolValue]) {
+                SentryStacktrace *sentryStacktrace = [thread stacktrace];
+                sentryStacktrace.snapshot = @(YES);
```

Comment:
Why, again, do we have to set the `snapshot` to `true`? Maybe we should also do that for ANRs

https://github.com/getsentry/sentry-cocoa/blob/8607e67a262a3e958b22e861bba7a57ec965b89c/Sources/Sentry/SentryANRTrackingIntegration.m#L56-L77

Reply:
It was explained in the spec PR, that means it's a synthetic exception, meaning the current state of the thread rather than the stack trace of a real exception, this changes the grouping behavior.
ANR on Android sets `snapshot=true`.

Reply:
https://github.com/getsentry/sentry-java/blob/649f1714ddbef043ded98bd491c19293d0a254f3/sentry-android-core/src/main/java/io/sentry/android/core/AnrIntegration.java#L88
Maybe raise an issue for Cocoa? this would be a breaking changing in grouping tho most likely.

Reply:
So we missed that for ANRs, then. Thanks